src/screens/AirdropDashboard/AirdropDashboardScreen.tsx

```diff
@@ -1,38 +1,97 @@
+import { connect } from 'react-redux';
 
 import { Header, ScreenTemplate } from 'app/components';
-import { Route, RootStackParams } from 'app/consts';
+import { Route, RootStackParams, Wallet } from 'app/consts';
 import { isAfterAirdrop } from 'app/helpers/airdrop';
+import { ApplicationState } from 'app/state';
+import {
+  checkSubscription,
+  CheckSubscriptionActionCreator,
+  subscribeWallet,
+  SubscribeWalletActionCreator,
+} from 'app/state/airdrop/actions';
+import * as airdropSelectors from 'app/state/airdrop/selectors';
 
-import { AirdropFinished, AirdropInProgress } from './components';
+import { AirdropFinished, AirdropInProgress, Footer } from './components';
 
 const i18n = require('../../../loc');
 
 interface Props {
+  wallets: Wallet[];
+  error: boolean;
+  isLoading: boolean;
+  checkSubscription: CheckSubscriptionActionCreator;
   navigation: CompositeNavigationProp<
     StackNavigationProp<RootStackParams, Route.MainTabStackNavigator>,
     StackNavigationProp<RootStackParams, Route.AirdropDashboard>
   >;
+  subscribedWallets: Wallet[];
+  availableWallets: Wallet[];
+  subscribeWallet: SubscribeWalletActionCreator;
 }
 
-// TODO: when implementing data fetching, add loading and error screens here.
-// loading: https://app.zeplin.io/project/5f0c8686b8151e82242dd409/screen/6005bdd8de723b5789229476
-// error: https://app.zeplin.io/project/5f0c8686b8151e82242dd409/screen/6005a32229c91b1a89551702
+export const AirdropDashboardScreen: FC<Props> = ({
+  navigation,
+  wallets,
+  checkSubscription,
+  subscribedWallets,
+  subscribeWallet,
+  availableWallets,
+  isLoading,
+  error,
+}) => {
+  useEffect(() => {
+    checkSubscription(wallets);
+  }, [checkSubscription, wallets]);
+  const airdropFinished = isAfterAirdrop();
 
-export const AirdropDashboardScreen: FC<Props> = ({ navigation }) => (
-  <ScreenTemplate header={<Header isBackArrow title={i18n.airdrop.title} />}>
-    <View style={styles.wrapper}>
-      {isAfterAirdrop() ? <AirdropFinished navigation={navigation} /> : <AirdropInProgress navigation={navigation} />}
-    </View>
-  </ScreenTemplate>
-);
+  return (
+    <ScreenTemplate
+      header={<Header isBackArrow title={i18n.airdrop.title} />}
+      footer={!airdropFinished && <Footer navigation={navigation} />}
+    >
+      <View style={styles.wrapper}>
+        {airdropFinished ? (
+          <AirdropFinished wallets={subscribedWallets} loading={isLoading} error={error} navigation={navigation} />
+        ) : (
+          <AirdropInProgress
+            loading={isLoading}
+            error={error}
+            availableWallets={availableWallets}
+            subscribedWallets={subscribedWallets}
+            subscribeWallet={subscribeWallet}
+          />
+        )}
+      </View>
+    </ScreenTemplate>
+  );
+};
 
-export default AirdropDashboardScreen;
+const mapStateToProps = (state: ApplicationState) => ({
+  wallets: airdropSelectors.airdropReadyWallets(state),
```

Comment:
unsued

Reply:
it is used in  line 48. We have to check which wallets are subscribed, right?

Reply:
ah right, didn't see

Reply:
user may have many wallets and not all of them will be suitable for airdrop, so we have to filter ones that are "ready"